soccer/KickEvaluator.cpp

```diff
@@ -0,0 +1,396 @@
+#include "KickEvaluator.hpp"
+#include <Utils.hpp>
+#include <Geometry2d/Util.hpp>
+
+#include <algorithm>
+#include <vector>
+#include <math.h>
+#include <cmath>
+
+static union {
+    float d;
+    struct {
+#ifdef LITTLE_ENDIAN
+        int j, i;
+#else
+        int i, j;
+#endif
+    } n;
+} _eco;
+
+// Floating point magic
+#define EXP_A (1048576 / 0.69314718055994530942)
+#define EXP_C 60801
+#define fast_exp(y) (_eco.n.i = EXP_A * (y) + (1072693248 - EXP_C), _eco.d)
```

Comment:
Oh woops I'm going insane. That implementation will only be better for calculating exp for literals(like e^100) Not for calculations. 
The benchmarks for that macro implementation don't seem much faster though (which is weird).
Did you see this one? https://codingforspeed.com/using-faster-exponential-approximation/
360x seems alot better than 
Slow exp time=306
vs
Fast exp time=286